src/rgw/rgw_rados.cc

```diff
@@ -3290,7 +3293,8 @@ class RGWRadosPutObj : public RGWHTTPStreamRWRequest::ReceiveCB
       JSONDecoder::decode_json("attrs", src_attrs, &jp);
 
       src_attrs.erase(RGW_ATTR_COMPRESSION);
-      src_attrs.erase(RGW_ATTR_MANIFEST); // not interested in original object layout
+      /* We need the manifest to recompute the ETag for verification */
+      manifest_bl = src_attrs[RGW_ATTR_MANIFEST];
```

Comment:
we should still erase RGW_ATTR_MANIFEST from src_attrs, because fetch_remote_obj() writes those directly to the new head object:
```c++
      auto iter = src_attrs.find(RGW_ATTR_MANIFEST);
      if (iter != src_attrs.end()) {
        manifest_bl = std::move(iter->second);
        src_attrs.erase(iter);
      }
```